joomla/joomla-cms
# [4.0] Correct css classes of mod_menu and the Cassiopeia template's css for menu at the "menu" module position

Pull Request for Issue # ?.

### Summary of Changes

This Pull Request (PR) removes the bootstrap class `flex-column` from the `ul` element of `mod_menu` and adds a BEM conforming class name to that element so this name can be used as selector in scss instead of `flex-column` in scss, and it corrects the scss for the header section of the Cassiopeia template so the menu looks as expected on the module position "menu", which currently is not the case.

Further removal of Bootstrap classes from the Casiopeia template is not subject of this PR but can be done in future.

It seems that the css has not been compiled after a recent change, so this PR shows more differences in templates/cassiopeia/css/template.css than those coming from this PR.

### Testing Instructions

Use current 4.0-dev branch with the default Cassiopeia frontent template and have a menu in module position `menu` which should be shown in horizontal direction, and some other menus in some side position shown in vertical direction.

I used blog testing data and moved the "Main Menu Blog" module to the `menu` position. In the module's advanced setting there is no class suffix defined for the module or the menu.

Check that the menu at the `menu` position (left beside the search box) is shown as expected with this PR applied.

Check that the look of all other things on the page has not changed after applying this PR.

Move the menu back to another position, e.g. `top-a` or `sidebar-right`, and check that it looks the same as without this PR at the same position.

### Expected result

See following picture with the menu marked with a red ellipse:
![after](https://user-images.githubusercontent.com/7413183/41190665-b917551e-6be3-11e8-85d0-167f29121bd3.png)

### Actual result

See following picture with the menu marked with a red ellipse:
![before](https://user-images.githubusercontent.com/7413183/41190668-be978752-6be3-11e8-9717-2d566c4cd620.png)

### Documentation Changes Required

None.
